### ISSUE TITLE
Example: Using imgui in a opengl scene

### ISSUE THREAD

**Issue by birgersp** [ocornut/imgui#1394]
Hi,

I've made a small OpenGL graphics engine (using SDL2). I want to use imgui "on top" to control stuff in the graphics engine. How can I do this?

Here's my attempt (scene is rendered, but no imgui windows are visible):

```
keepRunning = true;
while (keepRunning)
{
    while (SDL_PollEvent(&event) != 0)
    {
        handleEvent(event); // Handle event in game engine, may set keepRunning=false
        ImGui_ImplSdlGL2_ProcessEvent(&event); // Handle event in imgui
    }

    // Render game engine, binds buffers, draws triangles, draws all kinds of stuff (including a GUI)
    app.render();

    // Render imgui
    ImGui_ImplSdlGL2_NewFrame(sdlWindow);
    // 1. Show a simple window
    // Tip: if we don't call ImGui::Begin()/ImGui::End() the widgets appears in a window automatically called "Debug"
    {
        static float f = 0.0f;
        ImGui::Text("Hello, world!");
        ImGui::SliderFloat("float", &f, 0.0f, 1.0f);
        ImGui::ColorEdit3("clear color", (float*) &clear_color);
        if (ImGui::Button("Test Window")) show_test_window ^= 1;
        if (ImGui::Button("Another Window")) show_another_window ^= 1;
        ImGui::Text("Application average %.3f ms/frame (%.1f FPS)", 1000.0f / ImGui::GetIO().Framerate, ImGui::GetIO().Framerate);
    }

    // 2. Show another simple window, this time using an explicit Begin/End pair
    if (show_another_window)
    {
        ImGui::Begin("Another Window", &show_another_window);
        ImGui::Text("Hello from another window!");
        ImGui::End();
    }

    // 3. Show the ImGui test window. Most of the sample code is in ImGui::ShowTestWindow()
    if (show_test_window)
    {
        ImGui::SetNextWindowPos(ImVec2(650, 20), ImGuiCond_FirstUseEver);
        ImGui::ShowTestWindow(&show_test_window);
    }
    ImGui::Render();

    SDL_GL_SwapWindow(sdlWindow);
}
```

**Comment by ocornut** [comment]
- Try using the GL3 code. Mixing old style and new styles GL makes drivers unhappy.

- If you are using SDL_Renderer, they are mixed report about how it integrates with OpenGL draw calls, maybe need to flush something.
https://github.com/ocornut/imgui/search?q=SDL_Renderer&type=Issues&utf8=%E2%9C%93

- Otherwise comment out part of your code or imgui rendering code to find out what's causing a problem.

**Comment by birgersp** [comment]
Thanks for getting back to me! After half a day of attempting to figure out what I was doing wrong, it is now working as it should.
The problem was that vertex attribute arrays (that were enabled during rendering of my engine) were not disabled when invoking `ImGui::Render()`. Then, ImGui didn't render. I don't have to disable these vertex attribute arrays between using different shader programs, so I honestly didn't think it was a big deal to just keep them enabled (saving GL calls). Lesson learned.

imgui is awesome, thank you so much for all the effort!

**Comment by ocornut** [comment]
The OpenGL 2 binding has to way to disable/bind other arrays because those concepts don't exist in GL2, this is why you should use the GL3 code and I think it should have worked with it.
I would be interested if you can confirm that (if it works with the GL3 code _without_ the change in your engine code).

**Comment by birgersp** [comment]
So I've upgraded my application (including the ImGui implementation) to GL3. I was disabling all the vertex attributes, and "un-using" the shader programs before. But now I removed all of those (leaving attributes enabled, shader programs "in use" even though the rendering is finished, etc) and ImGui is working flawlessly.

I'm so grateful for ImGui at this point, was really not looking forward to GUI coding before I discovered ImGui but now it's just so easy (and automagically good-looking)!
